astro/src/content/quickstarts/quickstart-javascript-express-web.mdx

```diff
@@ -0,0 +1,320 @@
+
+1. Make sure to check that the `stateValue` that we stored in our cookie matches the `state` query parameters
+1. Request to the FusionAuth Token endpoint to exchange the authorization `code` returned from the Authorize endpoint and a `verifier` from our `userSession` cookie for an access token.
+1. Retrieves the user by passing the access token to FusionAuth. (This method does not use an API key, instead it uses a JSON Web Token (JWT) for authentication)
+1. Store access_token details in `userToken` cookie.
+1. Store retrieved user in `userDetails` cookie (note for our example this is not a secure cookie and JS can access).
+1. Redirect to `/account` route.
+
+`index.ts`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/index.ts"}
+  lang="ts"
+  tags="oauth-redirect"/>
+
+#### Account Route (protected)
+
+The /account page represents what a user sees when they’re logged into their Changebank account. A user that isn’t logged in, who tries to access this page, should just be taken to the home page.
+
+<Aside type="note">
+In a production application, we would validate the token signature, make sure it wasn't expired, and attempt to refresh it if it were expired.
+</Aside>
+
+`index.ts`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/index.ts"}
+  lang="ts"
+  tags="account"/>
+
+#### Logout Route
+
+<Aside type="tip">
+Click the `Logout` button and watch the browser first go to FusionAuth to log out the user, then return to your home page.
+</Aside>
+
+The last step is to implement logout. When you log a user out of an application, you'll take them to 
+FusionAuth's `/oauth2/logout` endpoint. After logging the user out, FusionAuth will redirect the user
+to your app's `/logout` endpoint, which you'll create now. This endpoint deletes any cookies that your
+application created, clearing the user's session.
+
+`index.ts`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/index.ts"}
+  lang="ts"
+  tags="logout"/>
+
+In the below code we check to see if the current user session has any information about the user, if not we redirect to the FusionAuth server to initiate the logout on the server. When this returns the user session cookie will now be empty and we can therefore forward back to our homepage, if we didn't add this check we would be stuck in an infinite loop between our Express server and FusionAuth (luckily unlike other providers you won't get billed per all :D)
+
+```ts
+if (userSession) {
+    res.redirect(302, `${fusionAuthURL}/oauth2/logout?client_id=${clientId}`);
+} else {
+    res.redirect(302, '/')
+}
+```
+
+## Bonus
+<Aside type="tip">
+As a bonus, try implementing the `/make-change` route using the `complete-application/templates/make-change.html` template, implementing the same validation as `/account` route.
```

Comment:
I thought we'd have to implement this in our quickstarts

Reply:
Great! But please don't forget to add those steps here.

Reply:
added if you would like to review